debug/log/log.go

```diff
@@ -0,0 +1,179 @@
+// Package log provides debug logging
+package log
+
+import (
+	"fmt"
+	"os"
+	"time"
+)
+
+var (
+	// DefaultLog logger
+	DefaultLog = NewLog()
+	// DefaultLevel is default log level
+	DefaultLevel = LevelInfo
+	// prefix for all messages
+	prefix string
+)
+
+// Log is event log
+type Log interface {
+	// Read reads log entries from the logger
+	Read(...ReadOption) []Record
+	// Write writes logs to logger
+	Write(...interface{})
```

Comment:
This should probably be ...Record so that metadata can be passed through but I know that's going to be an annoying change below so can always be done later.